### PR TITLE
nuttx/bits.h: support GENMASK and GENMASK_ULL

### PR DESCRIPTION
**Pull request by Donny9**
## Summary
do GENMASK and GENMASK_ULL marco.
Ref to https://github.com/torvalds/linux/blob/master/include/linux/bits.h
## Impact
add GENMASK and GENMASK_ULL marco.
## Testing
local test

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: include/nuttx/bits.h [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+#endif
+
+#if UINTPTR_MAX > UINT32_MAX
+  #define BITS_PER_LONG 64
+#else
+  #define BITS_PER_LONG 32
+#endif
+
+#ifndef BITS_PER_LONG_LONG
+  #define BITS_PER_LONG_LONG 64
```
Comment (pkarashchenko) [inline include/nuttx/bits.h:47]: ditto

File: include/nuttx/bits.h [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+#include <assert.h>
+#include <inttypes.h>
+#include <stdint.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#ifndef BITS_PER_BYTE
+  #define BITS_PER_BYTE 8
```
[the remark this answers is not in the document]
Comment (Donny9) [reply include/nuttx/bits.h]: done, and where define CHAR_BIT?

File: include/nuttx/bits.h [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+#endif
+
+#if UINTPTR_MAX > UINT32_MAX
+  #define BITS_PER_LONG 64
+#else
+  #define BITS_PER_LONG 32
```
[the remark this answers is not in the document]
Comment (Donny9) [reply include/nuttx/bits.h]: Done~

File: include/nuttx/bits.h (hunk 1 above)
Comment (Donny9) [reply include/nuttx/bits.h]: Done~

File: include/nuttx/bits.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (pkarashchenko) [reply include/nuttx/bits.h]: Here https://cplusplus.com/reference/climits/

**Review by acassis — APPROVED** [review]

**Pull request merged**